ipam/calico-ipam.go

```diff
@@ -222,24 +222,45 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	// Release the IP address by using the handle - which is workloadID.
-	workloadID, _, err := utils.GetIdentifiers(args)
+	// Release the IP address by using the handleID.
+	handleID, err := utils.GetHandleID(conf.Name, args.ContainerID)
 	if err != nil {
 		return err
 	}
 
-	logger := utils.CreateContextLogger(workloadID)
+	logger := utils.CreateContextLogger(handleID)
 
-	logger.Info("Releasing address using workloadID")
-	if err := calicoClient.IPAM().ReleaseByHandle(workloadID); err != nil {
+	logger.Info("Releasing address using handleID")
+	if err := calicoClient.IPAM().ReleaseByHandle(handleID); err != nil {
 		if _, ok := err.(errors.ErrorResourceDoesNotExist); ok {
-			logger.WithField("workloadId", workloadID).Warn("Asked to release address but it doesn't exist. Ignoring")
-			return nil
+			logger.Warn("Asked to release address but it doesn't exist. Ignoring")
```

Comment:
nit: you could/should reduce this down to a single branch by using `!ok` in the check above - then we don't need the "else" branch, because we'd just return in the error case.